### PR TITLE
[Keychron K7] Migrate to new folder structure

### PR DESCRIPTION
**Pull request by stdvar**
## Description

[Keychron K7] Migrate to new folder structure

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/k7/rgb/config.h [hunk 1]
```diff
@@ -33,13 +33,25 @@
...
+#define LED_CAPS_LOCK_PIN D4
 #define LED_PIN_ON_STATE  1
 
 /* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
 #define DEBOUNCE 5
 
 #define FORCE_NKRO
+
+// Backlight configuration
+#define BACKLIGHT_LEVELS 8
```
Comment (dexter93) [inline keyboards/keychron/k7/rgb/config.h]: afaik this does nothing in RGBMATRIX

File: keyboards/keychron/k7/rgb/rules.mk [hunk 2]
```diff
@@ -1,32 +1,34 @@
...
 
 # Custom RGB matrix handling
 RGB_MATRIX_ENABLE = yes
 RGB_MATRIX_DRIVER = SN32F24xB
+
+# project specific files
+SRC = ../../../drivers/led/sn32/matrix_sn32f24xx.c
+
+# Build ansi by default
+DEFAULT_FOLDER = keychron/k7/rgb/optical_ansi
```
Comment (dexter93) [inline keyboards/keychron/k7/rgb/rules.mk]: should probably favor the mech one, no?

File: keyboards/keychron/k7/rgb/rules.mk [hunk 3]
```diff
@@ -1,32 +1,34 @@
...
+MOUSEKEY_ENABLE         = no  # Mouse keys
+EXTRAKEY_ENABLE         = yes # Audio control and System control
+CONSOLE_ENABLE          = no  # Console for debug
+COMMAND_ENABLE          = no  # Commands for debug and configuration
+SLEEP_LED_ENABLE        = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE             = yes # USB Nkey Rollover
+AUDIO_ENABLE            = no
+RGBLIGHT_ENABLE         = no
+SERIAL_LINK_ENABLE      = no
+WAIT_FOR_USB            = yes
```
Comment (dexter93) [inline keyboards/keychron/k7/rgb/rules.mk]: shouldn't need to be ON by default, especially with the upcoming usb cleanup. Might need to look into this, probably not doing anything useful right now

File: keyboards/keychron/k7/rgb/rules.mk [hunk 4]
```diff
@@ -1,32 +1,34 @@
...
+BACKLIGHT_ENABLE        = no
+MAGIC_ENABLE            = yes
+MAGIC_KEYCODE_ENABLE    = yes
+BOOTMAGIC_ENABLE        = yes # Virtual DIP switch configuration
+MOUSEKEY_ENABLE         = no  # Mouse keys
+EXTRAKEY_ENABLE         = yes # Audio control and System control
+CONSOLE_ENABLE          = no  # Console for debug
+COMMAND_ENABLE          = no  # Commands for debug and configuration
+SLEEP_LED_ENABLE        = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE             = yes # USB Nkey Rollover
```
Comment (dexter93) [inline keyboards/keychron/k7/rgb/rules.mk:17]: need to discuss if we need this by default. Potential issues when the bt driver gets finished

File: quickbuild_whitelist [hunk 5]
```diff
@@ -4,5 +4,6 @@ redragon/k556
 keychron/k2/rgb
 keychron/k4/rgb/v1
 keychron/k4/rgb/v2
+keychron/k7/rgb
```
Comment (dexter93) [inline quickbuild_whitelist:7]: committing to testing regularly I see :stuck_out_tongue_closed_eyes:

File: keyboards/keychron/k7/rgb/config.h (hunk 1 above)
Comment (stdvar) [reply keyboards/keychron/k7/rgb/config.h]: k will remove

File: keyboards/keychron/k7/rgb/rules.mk (hunk 2 above)
Comment (stdvar) [reply keyboards/keychron/k7/rgb/rules.mk]: ok

File: quickbuild_whitelist (hunk 5 above)
Comment (stdvar) [reply quickbuild_whitelist:7]: yeah i dont mind for now. trying to stay in the loop of latest changes.

File: keyboards/keychron/k7/rgb/rules.mk (hunk 4 above)
Comment (stdvar) [reply keyboards/keychron/k7/rgb/rules.mk:17]: it is needed and useful. once bt is supported then we can change the default.

File: keyboards/keychron/k7/rgb/rules.mk (hunk 3 above)
Comment (stdvar) [reply keyboards/keychron/k7/rgb/rules.mk]: ok

**Pull request merged**